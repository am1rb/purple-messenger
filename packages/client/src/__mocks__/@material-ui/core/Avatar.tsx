import React from "react";

interface AvatarProps {
  src?: string;
}

function MuiAvatar({ src }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} />;
}

export default MuiAvatar;
