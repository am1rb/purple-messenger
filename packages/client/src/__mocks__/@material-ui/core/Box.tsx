import React from "react";

export interface BoxProps {
  children: React.ReactNode;
}

function MuiBox({ children }: BoxProps) {
  return <div data-testid="mock-mui-box">{children}</div>;
}

export default MuiBox;
