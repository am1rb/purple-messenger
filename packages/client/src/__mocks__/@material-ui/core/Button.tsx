import React from "react";

export interface ButtonProps {
  onClick?: () => void;
  children: React.ReactNode;
}

function MuiButton({ onClick, children }: ButtonProps) {
  return (
    <div data-testid="mock-mui-button" onClick={onClick}>
      {children}
    </div>
  );
}

export default MuiButton;
