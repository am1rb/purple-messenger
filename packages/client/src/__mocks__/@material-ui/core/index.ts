export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
