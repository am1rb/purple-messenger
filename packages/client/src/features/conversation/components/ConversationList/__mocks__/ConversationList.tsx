import React from "react";

interface ConversationListProps {
  selectedUsername: string | undefined;
}

function ConversationList({ selectedUsername }: ConversationListProps) {
  return (
    <div
      data-testid="mock-conversation-list"
      data-username={selectedUsername}
    />
  );
}

export default ConversationList;
