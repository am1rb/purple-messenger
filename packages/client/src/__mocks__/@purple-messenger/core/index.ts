import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
