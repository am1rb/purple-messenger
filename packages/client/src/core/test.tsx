import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import configureStore, { history } from "core/redux/configureStore";
import { State as RootState } from "core/redux/reducers";
import theme from "core/style/theme";
import { ThemeProvider } from "@material-ui/styles";

export function renderWithStore(
  ui: React.ReactElement,
  initStore: Partial<RootState>
) {
  const store = configureStore(initStore);

  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>{children}</ConnectedRouter>
        </Provider>
      </ThemeProvider>
    ),
  });
}
