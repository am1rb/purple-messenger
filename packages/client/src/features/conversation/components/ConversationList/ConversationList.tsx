import React, { memo, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import clsx from "clsx";
import {
  loadConversationList,
  unloadConversationList,
} from "@purple-messenger/core";
import { Box } from "@material-ui/core";
import { getConversationList } from "features/conversation/selectors";
import ConversationRow from "../ConversationRow";
import useStyles from "./styles";

interface Props {
  className: string;
}

function ConversationList({ className }: Props) {
  const classes = useStyles();
  const dispatch = useDispatch();
  const conversations = useSelector(getConversationList);

  useEffect(() => {
    dispatch(loadConversationList());
    return () => {
      dispatch(unloadConversationList());
    };
  }, []);

  return (
    <Box className={clsx(classes.root, className)} p={1}>
      {conversations.toArray().map(([, conversation]) => (
        <ConversationRow key={conversation.id} {...conversation} />
      ))}
    </Box>
  );
}

export default memo(ConversationList);
