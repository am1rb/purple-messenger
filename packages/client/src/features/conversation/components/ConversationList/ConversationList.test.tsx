import React from "react";
import { renderWithStore, InitialStore } from "core/test";
import ConversationList, { ConversationListProps } from "./ConversationList";
import { OrderedMap } from "immutable";
import {
  Conversation,
  AccountStatus,
  loadConversationList,
  unloadConversationList,
} from "@purple-messenger/core";

jest.mock("../ConversationRow");

const conversations: Conversation[] = [
  {
    id: 1,
    isTyping: false,
    friend: {
      id: 1,
      bio: "759 Ujipa Parkway",
      email: "potpe@ubdop.ml",
      firstName: "Maurice",
      lastName: "Underwood",
      image: "",
      status: AccountStatus.Online,
      username: "potpe",
    },
  },
  {
    id: 2,
    isTyping: true,
    friend: {
      id: 2,
      bio: "1573 Basog Square",
      email: "gizez@odudormar.do",
      firstName: "Alan",
      lastName: "Townsend",
      image: "",
      status: AccountStatus.Online,
      username: "alan",
    },
  },
];

const sharedProps: ConversationListProps = {
  username: undefined,
};

const initialStore: InitialStore = {
  conversation: {
    list: OrderedMap(
      conversations.map((conversation) => [
        conversation.friend.username,
        conversation,
      ])
    ),
  },
};

describe("The <ConversationList /> tests", () => {
  it("Should render rows properly", () => {
    const { container } = renderWithStore(
      <ConversationList {...sharedProps} />,
      initialStore
    );
    expect(container).toMatchSnapshot();
  });

  it("Should not select any conversation", () => {
    const { queryByTestId } = renderWithStore(
      <ConversationList {...sharedProps} />,
      initialStore
    );
    expect(
      queryByTestId("mock-conversation-row-selected")
    ).not.toBeInTheDocument();
  });

  it("Should select the first conversation", () => {
    const { getByTestId } = renderWithStore(
      <ConversationList
        {...sharedProps}
        username={conversations[0].friend.username}
      />,
      initialStore
    );
    expect(getByTestId("mock-conversation-row-selected")).toBeInTheDocument();
  });

  it("Should dispatch loadConversationList and unloadConversationList actions properly", () => {
    const { unmount } = renderWithStore(
      <ConversationList {...sharedProps} />,
      initialStore
    );
    expect(loadConversationList).toHaveBeenCalled();
    unmount();
    expect(unloadConversationList).toHaveBeenCalled();
  });
});
