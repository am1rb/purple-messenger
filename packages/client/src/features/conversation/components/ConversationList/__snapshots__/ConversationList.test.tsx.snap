// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ConversationList /> tests Should render rows properly 1`] = `
<div>
  <div
    data-testid="mock-mui-box"
  >
    <div
      data-testid="mock-conversation-row"
    />
    <div
      data-testid="mock-conversation-row"
    />
  </div>
</div>
`;
